packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1183,7 +1183,7 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   // Updates _imageStream to newStream, and moves the stream listener
   // registration from the old stream to the new stream (if a listener was
   // registered).
-  void _updateSourceStream(ImageStream newStream) {
+  void _updateSourceStream(ImageStream newStream, {bool updatedImage = false}) {
```

Comment:
instead of `updatedImage` maybe `forceGaplessPlayback` since that's what we are doing here?

Reply:
Also, I think this parameter should default to the old behavior and in places were you want the new behavior it should be explicitly set.

Reply:
The current problem with image is that when imageProvider is not updated, the imageCache no longer exists, and the images displayed on the interface will be cleared.
Adding the updatedImage parameter can clearly indicate that the imageProvider is updated, allowing the images displayed on the interface to be cleared.
I think this is different from the gaplessPlayback behavior.

The above content uses Google Translate.

Reply:
I think that the default clearing of displayed images is a defect of the Image widget. If explicit settings are required, the use of Images will become more troublesome because I need to change all the codes that use Images in the project.

Reply:
Image seems to be missing a builder, between loadingBuilder and frameBuilder. I see in the code that loadingBuilder is the loading callback and frameBuilder is the rendering callback.
However, on my Android device, after loading is complete, there will be a gap of about 100 milliseconds when the frameBuilder is called for the first time, and the Image will be blank at this time. There is no such problem on phones with higher configurations.

Reply:
I think when @/goderbauer said 
> this parameter should default to the old behavior and in places were you want the new behavior it should be explicitly set

he was talking about the private method `_updateSourceStream`. It's about making flags explicit to enhance the readability of implementation details. It has nothing to do with the default behavior of the Image widget as a whole. 

My suggestion is to change the method signatures to:
```
void _resolveImage({required bool imageUpdated});
void _updateSourceStream(ImageStream newStream, {required bool keepPreviousImage});
```

Where `keepPreviousImage = !imageUpdated || widget.gaplessPlayback`.

Reply:
> Image seems to be missing a builder, between loadingBuilder and frameBuilder. I see in the code that loadingBuilder is the loading callback and frameBuilder is the rendering callback. However, on my Android device, after loading is complete, there will be a gap of about 100 milliseconds when the frameBuilder is called for the first time, and the Image will be blank at this time. There is no such problem on phones with higher configurations.

Is this a different issue from what this PR is trying to solve?